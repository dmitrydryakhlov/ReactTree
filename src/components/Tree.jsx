import React from 'react';
import { connect } from 'react-redux';
import Node from './Node';
import { addNode, removeNode, changeNode } from '../actions/treeActions';

class Tree extends React.Component {

    render() {
        return <Node
            root={true}
            name={this.props.tree.name}
            children={this.props.tree.children}
            id={this.props.tree.id}
            addNode={this.props.addNode}
            removeNode={this.props.removeNode}
            changeNode={this.props.changeNode}
            level={0} />
    }
}

const mapStateToProps = (store) => {
    console.log(store);
    return {
        tree: store.tree
    }
}

const mapDispatchToProps = (dispatch) => ({
    addNode: (parentId, newNode) => dispatch(addNode(parentId, newNode)),
    removeNode: (id) => dispatch(removeNode(id)),
    changeNode: (id, newName) => dispatch(changeNode(id, newName))
})

export default connect(mapStateToProps, mapDispatchToProps)(Tree);

