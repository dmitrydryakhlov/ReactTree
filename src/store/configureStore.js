import { createStore, applyMiddleware } from 'redux';
import { rootReducer } from '../reducers/rootReducer';
import thunk from 'redux-thunk';
import { loadState, saveState } from '../utils';



const persistedStore = loadState();
const store = createStore(rootReducer, persistedStore, applyMiddleware(thunk));
store.subscribe(() => {
    saveState(store.getState());
})

export default store;