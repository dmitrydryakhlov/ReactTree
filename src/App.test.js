import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import Tree from './components/Tree';
import store from './store/configureStore';
import { act } from 'react-dom/test-utils';
import ReactTestUtils from 'react-dom/test-utils';

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
})

afterEach(() => {
  document.body.removeChild(container);
  container = null;
})

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('render correctly Tree component', () => {
  const rootElement = renderer.create(<Tree store={store} />).toJSON();
  expect(rootElement).toMatchSnapshot();
});

it('add Child', () => {
  act(() => {
    ReactDOM.render(<Tree store={store} />, container)
  })
  const addButton = container.querySelector('.addButton');

  act(() => {
    addButton.dispatchEvent(new MouseEvent('click', { bubbles: true }))
  });
  const label = container.querySelectorAll('span');

  act(() => {
    expect(label[0].textContent).toBe('root');
    expect(label[1].textContent).toBe('untitled');
  });

})

it('remove Node', () => {
  act(() => {
    ReactDOM.render(<Tree store={store} />, container);
  });
  let label = container.querySelectorAll('span');

  expect(label[0].textContent).toBe('root');
  expect(label[1].textContent).toBe('untitled');

  const removeButton = container.querySelectorAll('.removeButton');
  act(() => {
    removeButton[0].dispatchEvent(new MouseEvent('click', { bubbles: true }));
  })
  label = container.querySelectorAll('span');

  expect(label[0].textContent).toBe('root');
  expect(label[2]).toBe(undefined);
  expect(label[1]).toBe(undefined);
})

it('change Node ', () => {
  act(() => {
    ReactDOM.render(<Tree store={store} />, container);
  });
  let label = container.querySelectorAll('span');
  expect(label[0].textContent).toBe('root');
  expect(label[1]).toBe(undefined);

  const inputValue = container.querySelectorAll('input');
  inputValue[0].value = 'super';
  ReactTestUtils.Simulate.change(inputValue[0]);

  const changeButton = container.querySelectorAll('.changeButton');
  act(() => {
    changeButton[0].dispatchEvent(new MouseEvent('click', { bubbles: true }));
  })
  label = container.querySelectorAll('span');
  expect(label[0].textContent).toBe('super');
  expect(label[1]).toBe(undefined);
})
