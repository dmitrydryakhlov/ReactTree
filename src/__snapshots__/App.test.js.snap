// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly Tree component 1`] = `
<div>
  
  <span>
    root
  </span>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="addButton"
    onClick={[Function]}
  >
    +
  </button>
  <button
    className="changeButton"
    onClick={[Function]}
  >
    save
  </button>
</div>
`;
